crates/shim/src/asynchronous/publisher.rs

```diff
@@ -76,10 +152,22 @@ impl RemotePublisher {
         envelope.set_timestamp(timestamp()?);
         envelope.set_event(convert_to_any(event)?);
 
+        let item = Item {
+            ev: envelope.clone(),
+            ctx: ctx.clone(),
+            count: 0,
+        };
         let mut req = events::ForwardRequest::new();
         req.set_envelope(envelope);
 
-        self.client.forward(ctx, &req).await?;
+        if let Err(e) = self.client.lock().await.forward(ctx, &req).await {
```

Comment:
should you enqueue here, so the background queue process it?

Reply:
I means that if main thread client send fail , only this situation will push it in the queue, and try to resend it. So I use mutex and arc for main thread and `process queue ` task can use the client.  If all send to the queue ,the client will not need mutex and arc . This design is for the new event will not wait for the queue to deal. This design is same with goshim/publisher.go.